### PR TITLE
Replace generateSummaries with disableSummaries

### LINKED ISSUE #7605: Give ordering service the ability to disable summarizer client election
## Work Item
**Describe the outcome you expect**
Expose ordering service policy to disable clients' ability to elect a summarizer.

**Approach**

**Open questions**

**Additional context**
See https://github.com/microsoft/FluidFramework/issues/6260

**Acceptance criteria**

### PR DESCRIPTION
**Pull request by pleath**
In ContainerRuntime summary options, replace the generateSummaries flag with disableSummaries. If either this or the flag disableSummaries flag in ISummaryConfiguration is set on creation, the client will not participate in summarization or
summarizer client election. generateSummaries will be disabled for now,

### CODE REVIEW

**Comment by pleath** [comment]
Replaces https://github.com/microsoft/FluidFramework/pull/8170

File: packages/runtime/container-runtime/src/containerRuntime.ts [hunk 1]
```diff
@@ -2287,17 +2287,25 @@ export class ContainerRuntime extends TypedEventEmitter<IContainerRuntimeEvents>
         return this.pendingStateManager.getLocalState();
     }
 
+    /**
+     * @returns true if summaries are explicitly disabled for this ContainerRuntime, false otherwise
+     */
+    public summariesDisabled(): boolean {
+        return this.runtimeOptions.summaryOptions.disableSummaries === true ||
+            this.runtimeOptions.summaryOptions.summaryConfigOverrides?.disableSummaries === true;
```
Comment (vladsud) [inline packages/runtime/container-runtime/src/containerRuntime.ts:2295]: Adding disableSummaries  to ISummaryConfiguration?

**Comment by vladsud** [comment]
Might be worth adding breaking.md change.
Also I assume consumers will be forced to remove their usage of generateSummaries as TSC will bark at them, right? Trying to see how we ensure code outside of our repo stays clean(er)

**Review by vladsud — APPROVED** [review]

File: packages/runtime/container-runtime/src/containerRuntime.ts (hunk 1 above)
Comment (pleath) [reply packages/runtime/container-runtime/src/containerRuntime.ts:2295]: Yes, it was added there in a separate change.

**Comment by pleath** [comment]
> Might be worth adding breaking.md change. Also I assume consumers will be forced to remove their usage of generateSummaries as TSC will bark at them, right? Trying to see how we ensure code outside of our repo stays clean(er)

Would it be worth keeping generateSummaries around for the time being and deprecating it instead? The logic in container-runtime could easily map generateSummaries===false to disableSummaries===true.

**Comment by vladsud** [comment]
> Would it be worth keeping generateSummaries around for the time being and deprecating it instead? The logic in container-runtime could easily map generateSummaries===false to disableSummaries===true.

Probably yes. Note that host could provide that, and that means different layers (layer that updates slower than container runtime). So you will definitely have old host code and new container runtime code.
I think the more interesting thing - what happens if host today does not pass generateSummaries (which has default value of false), and now we are forcing summaries on them. I hope such cases do not exist really, but that's the only reason IMHO to thing and deal with back compat in some way

**Comment by pleath** [comment]
"I think the more interesting thing - what happens if host today does not pass generateSummaries (which has default value of false), and now we are forcing summaries on them." 

The current state of the ContainerRuntime ctor actually treats generateSummaries===true as the default:

        // Only create a SummaryManager if summaries are enabled and we are not the summarizer client
        if (this.runtimeOptions.summaryOptions.generateSummaries === false) {
            this._logger.sendTelemetryEvent({ eventName: "SummariesDisabled" });
        } else {
            [Set up for summarization.]

That's why I felt it was okay to replace generateSummaries with a flag that just disables summaries.

**Pull request merged**